Probe/src/main/data_collector.py

```diff
@@ -7,81 +8,120 @@
 import config
 
 
+class DataCollector():
 
 
-def dataCapture(noOfPackets = 2,filename = "f"):
-    ''' 
-    - Captures {noOfPackets} IPv6 data packets from all interfaces of the probe
-    - Stores the captured packets in a single .pcap file of name {filename}
-    - Annotates the .pcap file with hostname & date of capture
-    - Input: 
-        * noOfPackets : number of packets to be captured
-        * filename : the name by which the .pcap file is stored
-    - Output:
-        * none 
-    '''
+    def __init__(self,noOfPackets = 5) -> None:
 
-    logfilename = "capture_"+ datetime.datetime.now().strftime("%m-%d-%Y-%H-%M-%S")
-    logger = utils.getLogger(logfilename)
+        self.logfilename = "capture_"+ datetime.datetime.now().strftime("%m-%d-%Y-%H-%M-%S")
+        self.logger = utils.getLogger(self.logfilename)
 
-    capture_path = config.CAPTURE_PATH
+        self.capture_path = config.CAPTURE_PATH
+
+        if not os.path.exists(self.capture_path):  
+            try: 
+                os.makedirs(self.capture_path)
+            except OSError as e:
+                self.logger.error(e)
+                sys.exit(0)
+        
+        self.noOfPackets = noOfPackets
+        self.myhost = os.uname().nodename
 
-    if not os.path.exists(capture_path):  
         try: 
-            os.makedirs(capture_path)
+            with open(config.UID_FILE_PATH, 'r') as f:
+                self.uid = f.read()
+        except:
+            self.logger.error("No UID found, abborting data collection")
+            sys.exit(0)
+
+    def writeMetaData(self):
+        
+
+        posix = os.uname()
+        with open('/proc/meminfo') as f:
+            meminfo = f.read()
+        matched = re.search(r'^MemTotal:\s+(\d+)', meminfo)
+        if matched: 
+            mem_total_kB = int(matched.groups()[0])
+
+            
+        metadata = {
+            'UID': self.uid,
+            'sysname':posix.sysname,
+            'nodename':posix.nodename,
+            'release':posix.release,
+            'version':posix.version,
+            'machine':posix.machine,
+            'Number of packets captured per .pcap': self.noOfPackets,
+            'CPU Count':os.cpu_count(),
+            'total memory': mem_total_kB,
+            'all interfaces': scapy.get_if_list(),
+            'sniff interface': str(scapy.conf.iface),
+            'ip4 address': scapy.get_if_addr(scapy.conf.iface),
+            'mac address': scapy.get_if_hwaddr(scapy.conf.iface),
```

Comment:
@dhruvi29 let's make these metadata keys snake case. That way, it'll be much easier for future implementations.

Reply:
Done :+1: